### PR TITLE
[FIX] mail: add the ability to invite on mobile

### PR DESCRIPTION
**Pull request by phenix-factory**
Before this commit, it was impossible for a user to invite people to a channel
or a group DM.
This commit add the invite button on the chat windows header only on mobile.

Part of task-2634175

### CODE REVIEW

File: addons/mail/static/src/components/chat_window_header/chat_window_header.xml [hunk 1]
```diff
@@ -33,6 +33,16 @@
...
                     </t>
+                    <t t-if="chatWindow.thread and chatWindow.thread.hasInviteFeature and messaging.device.isMobile">
+                        <Popover popoverClass="'o_ThreadViewTopbar_invitePopover'" position="'bottom'">
+                            <button class="o_ThreadViewTopbar_inviteButton o_ThreadViewTopbar_button" title="Add users" t-on-click="chatWindow.onClickInviteButton">
+                                <i class="fa fa-lg fa-user-plus"/>
+                            </button>
+                            <t t-set="opened">
+                                <ChannelInvitationForm localId="chatWindow.threadView.channelInvitationForm.localId"/>
+                            </t>
+                        </Popover>
```
Comment (alexkuhn) [inline addons/mail/static/src/components/chat_window_header/chat_window_header.xml:44]: I'm not fond of using `Popover`, especially for newer code.
There's a PR in progress that adds proper `Popover` in models (and component that is a messaging component and doesn't use slots). I would much prefer to use that.

File: addons/mail/static/src/models/chat_window/chat_window.js [hunk 2]
```diff
@@ -138,6 +139,21 @@ function factory(dependencies) {
             this.update({ isMemberListOpened: true });
         }
 
+        /**
+         * Handles click on the "invite" button.
+         *
+         * @param {MouseEvent} ev
+         */
+        onClickInviteButton(ev) {
+            if (this.threadView.channelInvitationForm.component) {
```
Comment (alexkuhn) [inline addons/mail/static/src/models/chat_window/chat_window.js:148]: I would not guard at all, or at most `threadView.channelInvitationForm`.
I wouldn't take `.component`.
It's strange to allow click on invite when the component is no longer there, but I feel it doesn't matter as long as the intent is to click on invite button and the rest of code don't need a component.

**Review by alexkuhn — COMMENTED** [review]
Not a big deal to use popover in the meantime and then convert to newer component/model. Will wait for @seb-odoo assess whether it's worth it or not (depends on rough estimate to have newer Popover).

File: addons/mail/static/src/models/chat_window/chat_window.js (hunk 2 above)
Comment (phenix-factory) [reply addons/mail/static/src/models/chat_window/chat_window.js:148]: I reused the existing code, this guard was part of the function.

File: addons/mail/static/src/components/chat_window_header/chat_window_header.xml (hunk 1 above)
Comment (seb-odoo) [reply addons/mail/static/src/components/chat_window_header/chat_window_header.xml:44]: Popover is not really good in mobile indeed, better completely replace the thread view with the invite form in chat window (similar to what is done for the member list).

**Comment by phenix-factory** [comment]
@alexkuhn @seb-odoo This should be good now ?

**Review by seb-odoo — CHANGES_REQUESTED** [review]
Posting comments but I'll do them.

**Review by seb-odoo — COMMENTED** [review]
I made the changes above + had to make usability fixes (various CSS adaptations, better handling of message list/invite form conflicts, ...).

@alexkuhn could you final check it?

**Review by alexkuhn — APPROVED** [review]
Otherwise LGTM

**Comment by seb-odoo** [comment]
I noticed a crash when toggling the invitation form quickly (issue also possible in discuss) due to missing guards after await, I took to opportunity to add them here.

**Pull request closed without merging**